lib/source.c

```diff
@@ -109,6 +113,13 @@ tlog_source_read(struct tlog_source *source, struct tlog_pkt *pkt)
 
     assert(grc == TLOG_RC_OK || tlog_pkt_is_void(pkt));
     assert(tlog_source_is_valid(source));
+#ifndef NDEBUG
+    struct timespec diff;
```

Comment:
Could you please move the declaration of `diff` to the top of the function, joining `grc` there?

Reply:
Hi Nick, you want to take 2 assert in `NDEBUG` define. If I remove the `struct timespec diff`, compiler will ignore all asserts below?

Reply:
I'm sorry, I don't quite understand your question. I don't want to "take 2 assert in NDEBUG define", I'm just asking you to move the `struct timespec diff` declaration to the top of the `tlog_source_read` function, wrapping it in `#ifndef NDEBUG`, and putting it beside the `tlog_grc grc` declaration. That would make it conform to the tlog conventions better, as tlog usually has all declarations at the start of a function.

Reply:
Sorry I make you confused, I just want to make sure if  I write the code like: 
```
#ifndef NDEBUG
assert()
assert()
```
It will ignore all assert below. 

If I write something like: 
```
#ifndef NDEBUG
struct timespec diff
assert()
assert()
```
It will run when NDEBUG is not define